packages/pico-engine-core/src/index.js

```diff
@@ -287,12 +293,34 @@ module.exports = function(conf){
     var picoQ = PicoQueue(function(pico_id, job, callback){
         //now handle the next `job` on the pico queue
         if(job.type === "event"){
-            var event = job.event;
-            event.timestamp = new Date(event.timestamp);//convert from JSON string to date
-            processEvent(core, mkCTX({
-                event: event,
-                pico_id: pico_id
-            }), callback);
+            if (job.event.signedMessage) {
+                var secureEvent = job.event;
+                var signedMessage = secureEvent.signedMessage;
+                signedMessage = Uint8Array.from(_.toArray(signedMessage));
+                db.verifySignedMessage(signedMessage, secureEvent.eci, pico_id, function (err, verifiedMessage) {
+                    if (err || verifiedMessage === false) {
+                        var failedVerificationEvent = {};
+                        failedVerificationEvent.eci = secureEvent.eci;
+                        failedVerificationEvent.domain = "wrangler";
```

Comment:
Is there a way to fire off some sort of event when a message does fail to get signed, or if a verification fails? Or since it's moved to a engine function, will the user just check if it worked right after the function is called?

Reply:
Yes, you'll just check it in KRL. `engine:verifySignedMessage(verifyKey, message)` returns `false` if it failed to verify. So you'll probably do something like this:
```
pre {
  ...
  verifiedMessage = engine:engine:verifySignedMessage(...)
}
if verifiedMessage != false then
   noop()
fired {
   // now verifiedMessage is the original message string
} else {
   // didn't verify
   raise wrangler event "signature_verification_failed" ...
}
```

Reply:
Sweet, thanks! And is there a way in KRL to stringify a map? For example, if someone wants to sign a map object? Or is there a way to except either a string or a map in a krl function?

Reply:
Try  [encode](https://picolabs.atlassian.net/wiki/spaces/docs/pages/1189580/Universal+Operators#UniversalOperators-encode()).

Reply:
That worked! Thanks!